app/views/vaccinations/_patients_with_actions.html.erb

```diff
@@ -1,26 +1,28 @@
+<h2 class="nhsuk-heading-s nhsuk-u-margin-0 app-visually-hidden-from-desktop"><%= caption %></h2>
+
 <action-table class="nhsuk-grid-row" sort="name">
   <div class="nhsuk-grid-column-one-quarter">
     <%= render AppPatientTableFilterComponent.new(tab_id: id, filter_actions: true) %>
   </div>
   <div class="nhsuk-grid-column-three-quarters" id="patients-<%= id %>">
-    <% if patient_sessions.any? %>
-      <table class="nhsuk-table app-table--patients">
-        <caption class="nhsuk-table__caption"><%= t("states.action_needed.title") %></caption>
-        <thead class="nhsuk-table__head">
-          <tr class="nhsuk-table__row">
-            <th class="nhsuk-table__header" data-col="name">Name</th>
-            <th class="nhsuk-table__header app-u-nowrap" data-col="dob">Date of birth</th>
-            <th class="nhsuk-table__header" data-col="action" style="width: 200px">Action needed</th>
-          </tr>
-        </thead>
-        <tbody class="nhsuk-table__body">
-          <% patient_sessions.each do |patient_session| %>
-            <%= render "patient_row", patient_session: %>
-          <% end %>
-        </tbody>
-      </table>
-    <% else %>
-      <%= render AppEmptyListComponent.new %>
-    <% end %>
+    <table class="nhsuk-table app-table--patients">
+      <thead class="nhsuk-table__head">
+        <tr class="nhsuk-table__row">
+          <th class="nhsuk-table__header" data-col="name">Name</th>
+          <th class="nhsuk-table__header app-u-nowrap" data-col="dob">Date of birth</th>
+          <th class="nhsuk-table__header" data-col="action" style="width: 200px">Action needed</th>
+        </tr>
+      </thead>
+      <tbody class="nhsuk-table__body">
+        <% patient_sessions.each do |patient_session| %>
+          <%= render "patient_row", patient_session: %>
+        <% end %>
+      </tbody>
+    </table>
+    <action-table-no-results>
+      <%= render AppEmptyListComponent.new(
+        text: t("table.no_filtered_results")
+      ) %>
+    </action-table-no-results>
   </div>
-</action-table>
+</action-table>
```

Comment:
@paulrobertlloyd can you configure your editor to include a trailing newline at the end of each file? That's why GitHub is showing a red circle with a dash in the middle